consensus/src/simplex/actors/voter/state.rs

```diff
@@ -197,20 +205,21 @@ impl<E: Clock + Rng + CryptoRng + Metrics, S: Scheme<D>, L: ElectorConfig<S>, D:
         let nullify = Nullify::sign::<D>(&self.scheme, &self.namespace, Rnd::new(self.epoch, view));
 
         // If was retry, we need to get entry certificates for the previous view
-        if !retry || view.previous().is_none_or(|v| v == GENESIS_VIEW) {
+        let entry_view = view.previous().unwrap_or(GENESIS_VIEW);
+        if !retry || entry_view == GENESIS_VIEW {
             return (retry, nullify, None);
         }
-        let entry_view = view.previous().expect("checked to be non-zero above");
 
-        // Try to construct entry certificates for the previous view
-        // Prefer the strongest proof available so lagging replicas can re-enter quickly.
+        // Get the certificate for the previous view. Prefer finalizations since they are the
+        // strongest proof available. Prefer nullifications over notarizations since they do not
+        // require certification to move to the next view.
         #[allow(clippy::option_if_let_else)]
         let cert = if let Some(finalization) = self.finalization(entry_view).cloned() {
             Some(Certificate::Finalization(finalization))
-        } else if let Some(notarization) = self.notarization(entry_view).cloned() {
-            Some(Certificate::Notarization(notarization))
         } else if let Some(nullification) = self.nullification(entry_view).cloned() {
             Some(Certificate::Nullification(nullification))
```

Comment:
nit: explain we prefer nullifications because they `overwrite` a notarization of uncertified data